bftengine/src/preprocessor/PreProcessor.cpp

```diff
@@ -394,46 +405,37 @@ PreProcessor::PreProcessor(shared_ptr<MsgsCommunicator> &msgsCommunicator,
-      // TODO - this is a temporary in-place configuration - should be replaced with a future task to take it from
+      dynamicSubpoolsConfig_{YAML::Load(myReplica_.getReplicaConfig().get<std::string>(
+                                            "concord.preprocessor.multiSizedPool.subpoolsConfig", "[]"))
+                                 .as<std::vector<concordUtil::MultiSizeBufferPool::SubpoolConfig>>()},
+      subpoolsConfig_{dynamicSubpoolsConfig_.size() > 0 ? dynamicSubpoolsConfig_ : defaultSubpoolsConfig_},
       // replica config
-      bufferPool_{myReplica_.getReplicaConfig().enablePreProcessorMemoryPool
-                      ? std::make_unique<concordUtil::MultiSizeBufferPool>(
-                            "PreProcessor::memoryPool",
-                            timers_,
-                            subpoolsConfig_,
-                            concordUtil::MultiSizeBufferPool::Config{// maxAllocatedBytes
-                                                                     10ULL * (1 << 30),
-                                                                     // WarnOnSuspectedLeak
-                                                                     true,
-                                                                     // purgeEvaluationFrequency
-                                                                     0,
-                                                                     // statusReportFrequency
-                                                                     60,
-                                                                     // histogramsReportFrequency
-                                                                     60,
-                                                                     // metricsReportFrequency
-                                                                     5,
-                                                                     // enableStatusReportChangesOnly
-                                                                     true,
-                                                                     // subpoolSelectorEvaluationNumRetriesMinThreshold
-                                                                     3,
-                                                                     // subpoolSelectorEvaluationNumRetriesMaxThreshold
-                                                                     10})
-                      : nullptr},
+      bufferPool_{
+          myReplica_.getReplicaConfig().enablePreProcessorMemoryPool
+              ? std::make_unique<concordUtil::MultiSizeBufferPool>(
+                    "PreProcessor::memoryPool",
+                    timers_,
+                    subpoolsConfig_,
+                    concordUtil::MultiSizeBufferPool::Config{
+                        myReplica_.getReplicaConfig().get<uint64_t>(
+                            "concord.preprocessor.multiSizedPool.maxAllocatedBytes", 10ULL * (1 << 30)),
```

Comment:
We didnt talk about real config.
please keep all the same but set maxAllocatedBytesto 10GB and enableStatusReportChangesOnly=false

Reply:
Fixed enableStatusReportChangesOnly,
maxAllocatedBytesto is already 10GB...